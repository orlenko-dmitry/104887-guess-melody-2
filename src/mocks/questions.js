const questions = [
  {
    "type": `artist`,
    "song": {
      "artist": `The Beatles`,
      "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/6/6d/National_anthem_of_Ukraine%2C_instrumental.oga/National_anthem_of_Ukraine%2C_instrumental.oga.mp3`
    },
    "answers": [
      {
        "id": 1,
        "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
        "artist": `The Beatles`
      },
      {
        "id": 2,
        "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
        "artist": `Oasis`
      },
      {
        "id": 3,
        "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
        "artist": `The Kinks`
      }
    ]
  },
  {
    "type": `genre`,
    "genre": `reggae`,
    "answers": [
      {
        "id": 1,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/6/6d/National_anthem_of_Ukraine%2C_instrumental.oga/National_anthem_of_Ukraine%2C_instrumental.oga.mp3`,
        "genre": `reggae`
      },
      {
        "id": 2,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b4/United_States_Navy_Band_-_O_Canada.ogg/United_States_Navy_Band_-_O_Canada.ogg.mp3`,
        "genre": `electronic`
      },
      {
        "id": 3,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/3/30/La_Marseillaise.ogg/La_Marseillaise.ogg.mp3`,
        "genre": `country`
      },
      {
        "id": 4,
        "src": `https://upload.wikimedia.org/wikipedia/commons/transcoded/c/c8/Marcha_Real-Royal_March_by_US_Navy_Band.ogg/Marcha_Real-Royal_March_by_US_Navy_Band.ogg.mp3`,
        "genre": `reggae`
      }
    ]
  }
];

export default questions;
