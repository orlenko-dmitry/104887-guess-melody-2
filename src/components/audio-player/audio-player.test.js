import React from 'react';
import {render} from '@testing-library/react';

import AudioPlayer from './audio-player.jsx';
import questions from '../../mocks/questions.js';

const {song: {src}} = questions[0];

it(`AudioPlayer renders correctly`, () => {
  const {container} = render(
      <AudioPlayer
        src={src}
        isPlaying={false}
        onPlayButtonClick={() => {}}
      />
  );

  expect(container).toMatchSnapshot();
});
