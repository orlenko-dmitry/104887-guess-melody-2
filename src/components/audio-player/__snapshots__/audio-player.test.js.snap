// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer renders correctly 1`] = `
<button
  class="track__button track__button--play"
  disabled=""
  type="button"
/>
`;
